sprite.cpp

```diff
@@ -8,8 +8,15 @@
 
 using std::get;
 
-Sprite::Sprite(const Texture *texture, const Point &home) 
-	: m_texture(texture), m_home(home), m_relpos(0.0f, 0.0f), m_size(cfg[Cfg::SpriteSize] / 1000.0f) { }
+Sprite::Sprite(const Texture *texture, const Point &home, const bool has_trail)
+	: m_texture(texture), m_home(home), m_relpos(0.0f, 0.0f), m_size(cfg[Cfg::SpriteSize] / 1000.0f)
+{
+	if (has_trail && round(cfg[Cfg::TrailLength]) > 1) {
+		for (int i = 0; i < (int)(round(std::clamp(cfg[Cfg::TrailLength], 1.0f, cfg[Cfg::MaxTrailCount] / cfg[Cfg::SpriteCount]) - 1) * max(round(cfg[Cfg::TrailSpace]), 1)); i++) {
```

Comment:
I know I'm not exactly a good example for not writing way-too-long single lines of code, but this is a pretty beefy for loop - it wraps three whole times in the preview here. It would help a lot to break out into some variables so it's clearer what each of these calculations is doing and saves some space.

Reply:
Unsurprisingly, I agree. There's far too much happening in this one line.

Reply:
These have been cleaned up some.